In [2]:
import pandas as pd
import numpy as np


In [3]:
df = pd.read_csv("student_performance.csv")
df


student_id  weekly_self_study_hours  attendance_percentage  \
0            1                     18.5                   95.6   
1            2                     14.0                   80.0   
2            3                     19.5                   86.3   
3            4                     25.7                   70.2   
4            5                     13.4                   81.9   
..         ...                      ...                    ...   
94          95                     12.3                   70.9   
95          96                      4.8                   79.9   
96          97                     17.1                   76.7   
97          98                     16.8                   95.7   
98          99                     15.0                   80.7   

    class_participation  total_score grade  
0                   3.8         97.9     A  
1                   2.5         83.9     B  
2                   5.3        100.0     A  
3                   7.0        100.0     A  
4                   6.9         92.0     A  
..                  ...          ...   ...  
94                  8.5         72.0     B  
95                  4.2         73.6     B  
96                  7.6         67.3     C  
97                  5.3         73.1     B  
98                  6.0        100.0     A  

[99 rows x 6 columns]

In [4]:
df.head()
df.tail()


student_id  weekly_self_study_hours  attendance_percentage  \
94          95                     12.3                   70.9   
95          96                      4.8                   79.9   
96          97                     17.1                   76.7   
97          98                     16.8                   95.7   
98          99                     15.0                   80.7   

    class_participation  total_score grade  
94                  8.5         72.0     B  
95                  4.2         73.6     B  
96                  7.6         67.3     C  
97                  5.3         73.1     B  
98                  6.0        100.0     A

In [5]:
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99 entries, 0 to 98
Data columns (total 6 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   student_id               99 non-null     int64  
 1   weekly_self_study_hours  99 non-null     float64
 2   attendance_percentage    99 non-null     float64
 3   class_participation      99 non-null     float64
 4   total_score              99 non-null     float64
 5   grade                    99 non-null     object 
dtypes: float64(4), int64(1), object(1)
memory usage: 4.8+ KB


In [9]:
df.shape

(99, 6)

The dataset contains 99 rows and 6 columns.  
Data types include int, float and object.


### Dataset Information
The dataset has 99 records and 6 features. The grade column is categorical while other columns are numerical.


### Dataset Structure Observation

The dataset contains 99 entries and 6 columns.  
The numerical columns are: student_id, weekly_self_study_hours, attendance_percentage, class_participation, and total_score.  
The categorical column is: grade.  

The dataset does not contain any missing values, which means the data quality is good for machine learning.


In [10]:
df.describe()


student_id  weekly_self_study_hours  attendance_percentage  \
count   99.000000                99.000000              99.000000   
mean    50.000000                14.321212              83.798990   
std     28.722813                 6.305282               8.882515   
min      1.000000                 0.000000              64.400000   
25%     25.500000                10.800000              78.150000   
50%     50.000000                14.200000              83.700000   
75%     74.500000                18.050000              89.850000   
max     99.000000                28.000000             100.000000   

       class_participation  total_score  
count            99.000000    99.000000  
mean              6.047475    82.878788  
std               1.808503    15.338642  
min               2.500000    47.400000  
25%               4.600000    71.100000  
50%               6.000000    84.000000  
75%               7.600000    97.700000  
max               9.700000   100.000000

### Statistical Summary Observation

The average weekly self study hours of students is approximately 14.32 hours.  
The average attendance percentage is about 83.79%.  
The average class participation score is around 6.04.  
The average total score of students is 82.87.

The minimum total score is 47.4 and the maximum total score is 100, showing good performance variation among students.


In [11]:
df['grade'].value_counts()


grade
A    48
B    27
C    19
D     5
Name: count, dtype: int64

### Grade Distribution Observation

The grade distribution shows that:

- Grade A: 48 students  
- Grade B: 27 students  
- Grade C: 19 students  
- Grade D: 5 students  

This indicates that the dataset is imbalanced, as the number of students in grade A is much higher compared to grade D. Data imbalance can affect machine learning model performance and should be handled using proper techniques.


### Target and Input Features

The target variable in this dataset is **grade**, as it represents the outcome to be predicted.

The input features are:
- weekly_self_study_hours
- attendance_percentage
- class_participation
- total_score


### Conclusion

The Student Performance dataset contains clean numerical data with one categorical target variable.  
The dataset has no missing values and is suitable for machine learning classification tasks after minor preprocessing.  
However, data imbalance should be considered during model training.
